mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoDeleteMethodGenerator.java

```diff
@@ -120,25 +122,39 @@ public DaoDeleteMethodGenerator(
         return Optional.empty();
       }
       entityDefinition = context.getEntityFactory().getDefinition(entityElement);
-      List<TypeName> primaryKeyTypes =
-          entityDefinition.getPrimaryKey().stream()
-              .map(d -> d.getType().asTypeName())
-              .collect(Collectors.toList());
-      List<TypeName> parameterTypes =
-          parameters.stream().map(p -> TypeName.get(p.asType())).collect(Collectors.toList());
-      // Note that we only check the types: we allow the names to be different.
-      if (parameterTypes.size() < primaryKeyTypes.size()
-          || !primaryKeyTypes.equals(parameterTypes.subList(0, primaryKeyTypes.size()))) {
-        context
-            .getMessager()
-            .error(
-                methodElement,
-                "Invalid parameter list: %s methods that do not operate on an entity instance "
-                    + "must take the partition key components in the exact order "
-                    + "(expected PK of %s: %s)",
-                Delete.class.getSimpleName(),
-                entityElement.getSimpleName(),
-                primaryKeyTypes);
+      // if a custom if clause is provided, the whole primary key must also be provided.
+      List<? extends VariableElement> keyParameters = parameters;
+      if (!annotation.customIfClause().isEmpty()) {
+        if (keyParameters.size() < entityDefinition.getPrimaryKey().size()) {
```

Comment:
This gave me pause because the check is not accurate, we don't know how many placeholders the custom IF clause has.
But it's more of a guard so that we can perform the `subList` call in the other branch; if this check succeeds it's indeed a definitive mistake (even though there could still be something wrong if the check fails, but that will be caught later).

Reply:
That's right.  It's not legal to not provide the full primary key and use a custom if clause.

So there's possibly the case where enough parameters are provided to fulfill the number of primary key parameters, but some of those parameters were meant for the custom if clause.  In this case it's likely it won't match the primary key definition and you'll get an error which explains that he parameters don't match the primary key, which is technically correct.  If they do somehow match, there's not much we can do at compile time (I think?).  In any case, you'll get an error at run time.